tests/ReactionsApiTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+use Rogue\Models\Photo;
+
+class ReactionsApiTest extends TestCase
+{
+    /*
+     * Base URL for the Api.
+     */
+    protected $reactionsApiUrl = 'api/v2/reactions';
+
+    /**
+     * Test that the POST /reactions request creates a reaction for a reactionable.
+     * Also test that when POST /reactions is hit again by the same user for the same reactionable,
+     * the reaction is soft deleted.
+     *
+     * POST /reactions
+     * @return void
+     */
+    public function testPostingAndSoftDeleteForReaction()
+    {
+        // Create a photo to react to.
+        $photo = factory(Photo::class)->create();
+
+        $reaction = $this->createReactionRequest(str_random(24), $photo->id, 'photo');
+
+        $this->json('POST', $this->reactionsApiUrl, $reaction);
```

Comment:
This is subjective, but I think it makes sense to inline both of these to reduce the amount of jumping around you need to do when reading the test case. In general I always try to make test code as explicit as possible so it's easy to follow exactly what steps are happening in what order.

```php
$this->json('POST', 'api/v2/reactions', [
    'northstar_id' => $this->faker->uuid,
    'reactionable_id' => $photo->id,
    'reactionable_type' => 'photo',
]);
```